src/main/java/com/ericsson/ei/frontend/utils/BackEndInstanceFileUtils.java

```diff
@@ -70,16 +86,34 @@ public void dumpJsonArray(JsonArray jsonArrayToDump) {
     }
 
     private void ensureValidFile() throws IOException {
-        if (!(new File(eiInstancesPath).isFile())) {
-            LOG.error("File does not exist! Trying to creat file.");
-            Files.createFile(Paths.get(eiInstancesPath));
-            Files.write(Paths.get(eiInstancesPath), "[]".getBytes());
-            return;
+        try {
+            if (!(new File(eiInstancesPath).isFile())) {
+                createFileWithDirs();
+                return;
+            }
+
+            if (!fileContainsJsonArray()) {
+                LOG.error("File does not contain valid json! JSON:'" + new String(Files.readAllBytes(Paths.get(eiInstancesPath))) + "'.");
```

Comment:
If the File does not contain valid JSON then maybe we should also do a System.exit.
Note that if file is empty "" we could add just [] and it would be valid. This is if user creates an empty file or manually deletes all content.